packages/ui/src/components/PropertiesModal/PropertiesModal.tsx

```diff
@@ -0,0 +1,76 @@
+import { Modal } from '@patternfly/react-core';
+import { Caption, Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { FunctionComponent } from 'react';
+import { camelComponentToTable, kameletToTable } from '../../camel-utils/camel-to-table.adapter';
+import { CatalogKind, ICamelComponentDefinition, ICamelProcessorDefinition, IKameletDefinition } from '../../models';
+import { ITile } from '../Catalog';
+import { EmptyTableState } from './EmptyTableState';
+import { IPropertiesTable, PropertiesHeaders } from './PropertiesTable.models';
+
+interface IPropertiesModalProps {
+  tile: ITile;
+  onClose: () => void;
+  isModalOpen: boolean;
+}
+
+export const PropertiesModal: FunctionComponent<IPropertiesModalProps> = (props) => {
+  let table: IPropertiesTable;
+  switch (props.tile.type) {
+    case CatalogKind.Component:
+      table = camelComponentToTable(props.tile.rawObject as ICamelComponentDefinition);
+      break;
+    case CatalogKind.Processor:
+      table = camelComponentToTable(props.tile.rawObject as ICamelProcessorDefinition);
+      break;
```

Comment:
```suggestion
    case CatalogKind.Component:
    case CatalogKind.Processor:
      table = camelComponentToTable(props.tile.rawObject as ICamelProcessorDefinition);
      break;
```

Reply:
I don't want to add them together since one uses `ICamelComponentDefinition` and the other one uses `ICamelProcessorDefinition`. They are the same now but in the future, they can be different, they can't? ( if not, we can also generalize it in the models package )

Reply:
When the situation come, we can split it again I think, but is not a major thing, we can keep it as you have it now

Reply:
i would like to change the function `camelComponentToTable` completely in the future to have only properties as an input because CamelComponentDefinition has 3 types of properties ( properties, headers, componentProperties ) and each one should have a different table.